kafka_consumer/datadog_checks/kafka_consumer/new_kafka_consumer.py

```diff
@@ -488,3 +468,79 @@ def _collect_broker_metadata(self):
     def batchify(iterable, batch_size):
         iterable = list(iterable)
         return (iterable[i : i + batch_size] for i in range(0, len(iterable), batch_size))
+
+    # FIXME: This is using a workaround to skip socket wakeup, which causes blocking
+    # (see https://github.com/dpkp/kafka-python/issues/2286).
+    # Once https://github.com/dpkp/kafka-python/pull/2335 is merged in, we can use the official
+    # implementation for this function instead.
+    def _send_request_to_node(self, node_id, request, wakeup=True):
+        while not self.kafka_client._client.ready(node_id):
+            # poll until the connection to broker is ready, otherwise send()
+            # will fail with NodeNotReadyError
+            self.kafka_client._client.poll()
+        return self.kafka_client._client.send(node_id, request, wakeup=wakeup)
+
+    def _list_consumer_groups_send_request(self, broker_id):
+        kafka_version = self.kafka_client._matching_api_version(ListGroupsRequest)
+        if kafka_version <= 2:
+            request = ListGroupsRequest[kafka_version]()
+        else:
+            raise NotImplementedError(
+                "Support for ListGroupsRequest_v{} has not yet been added to KafkaAdminClient.".format(kafka_version)
+            )
+        # Disable wakeup when sending request to prevent blocking send requests
+        return self._send_request_to_node(broker_id, request, wakeup=False)
+
+    def _find_coordinator_id_send_request(self, group_id):
+        """Send a FindCoordinatorRequest to a broker.
+        :param group_id: The consumer group ID. This is typically the group
+            name as a string.
+        :return: A message future
+        """
+        # TODO add support for dynamically picking version of
+        # GroupCoordinatorRequest which was renamed to FindCoordinatorRequest.
+        # When I experimented with this, the coordinator value returned in
+        # GroupCoordinatorResponse_v1 didn't match the value returned by
+        # GroupCoordinatorResponse_v0 and I couldn't figure out why.
+        version = 0
+        # version = self._matching_api_version(GroupCoordinatorRequest)
+        if version <= 0:
+            request = GroupCoordinatorRequest[version](group_id)
```

Comment:
Are these lines from `kafka-python`? If we initialize `version = 0`, wouldn't `if version <= 0` always be true? Why is line 506 commented out?

Reply:
Yup these lines are exactly from the `kafka-python` library, `TODO` and all: https://github.com/dpkp/kafka-python/blob/4d598055dab7da99e41bfcceffa8462b32931cdd/kafka/admin/client.py#L293-L313. I can fix this to make it look better, although originally I wasn't sure if I should modify their library code.

Reply:
The unfortunate thing is this library is basically deprecated, so that `TODO` will likely never be added 😢